CDCgov/prime-reportstream
# Facilities call done with rest-hooks

This PR removes the old `fetch()` call to get facilities and configures the `FacilityResource` class with a custom config. All we have to do now is call `FacilityResource.list()` and pass in the `reportId`.

Test Steps:
1. pull `kevinhaube/rest-hooks-facilities`
2. if we're on `npm`, use `npm install`, else `yarn install`
3. serve the application
4. navigate to a `/report-details` page
5. ensure facilities load and populate the table properly
6. check Network and Console for any errors or warnings
7. :moneybag: profit :moneybag: 

## Changes
- `FacilityResource.ts` set up with custom `listUrl` override.
- `FacilitiesTable` updated to use `.list()` call instead of old `fetch()` method

## Checklist

### Testing

### Process

## Fixes
- #2365 

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
